resources/announcements.py

```diff
@@ -0,0 +1,137 @@
+                ret = Announcement_Schema().dump(announcement).data
+                return (ret,200,headers)
+            else:
+                return (not_found,404,headers)
+
+    def delete(self,announcement_id):
+        """
+        For DELETE request for specific id
+        """
+        user_status,user = has_admin_privileges()
+        if user_status == "no_auth_token":
+            return (bad_request,400,headers)
+
+        if user_status == "not_logged_in":
+            return (unauthorised,401,headers)
+
+        if user_status == True:
+            try:
+                g.session.query(g.Base.classes.announcements).filter(g.Base.classes.announcements.id == announcement_id).one()
+                g.session.query(g.Base.classes.announcements).filter(g.Base.classes.announcements.id == announcement_id).delete()
+                return ("",204,headers)
+            except NoResultFound:
+                return (not_found,404,headers)
+            except:
+                print(type(err))
+                print(err)
+                return (internal_server_error, 500, headers)
+        else:
+            return (forbidden,403,headers)
+class Announcements_CR(Resource):
+    """
+    For GET PUT and DELETE for specific announcement
+    """
+    def get(self):
+        """
+        Getting all the announcements by hitting /announcements endpoint without any specific id
+        """
+        try:
+            all_announcements = g.session.query(g.Base.classes.announcements).all()
+            ret = Announcement_Schema(many = True).dump(all_announcements).data
+            return (ret,200,headers)
+        except:
+            return (internal_server_error,500,headers)
+
+    def post(self):
+        """
+        For POST request. Checks for bad JSON from request, checks if data types in request JSON are correct, Checks user auth status and if he's logged in.
+        """
+        try:
+            data = request.get_json(force=True)
+        except BadRequest:
+            return (bad_request,400,headers)
+
+        #data validation
+        if Announcement_Schema().validate(data):
+            return (unprocessable_entity,422,headers)
+
+        #check if user has admin privileges
+        user_status,user = has_admin_privileges()
+        if user_status == "no_auth_token":
+            return (bad_request,400,headers)
+
+        if user_status == "not_logged_in":
+            return (unauthorised,401,headers)
+
+        if user_status == True:
+            Announcements = g.Base.classes.announcements
+            try:
+                new_announcement = Announcements(
+                                                    title = data["title"],
+                                                    description = data["description"],
+                                                    pinned = data["pinned"],
+                                                    updated_at = datetime.now(),
```

Comment:
Think about using pendulum library for datetime stuff https://pendulum.eustace.io/

Reply:
The pendulum library doesn't add anything specific that is not being solved the inbuilt datetime library. We should try to keep the memory footprint of the API to a minimum.
https://aboutsimon.com/blog/2016/08/04/datetime-vs-Arrow-vs-Pendulum-vs-Delorean-vs-udatetime.html

Reply:
That's a good point, I agree.